cmd/kubeadm/app/util/config/initconfiguration.go

```diff
@@ -85,9 +86,12 @@ func SetBootstrapTokensDynamicDefaults(cfg *[]kubeadmapi.BootstrapToken) error {
 // SetNodeRegistrationDynamicDefaults checks and sets configuration values for the NodeRegistration object
 func SetNodeRegistrationDynamicDefaults(cfg *kubeadmapi.NodeRegistrationOptions, ControlPlaneTaint bool) error {
 	var err error
-	cfg.Name, err = nodeutil.GetHostname(cfg.Name)
-	if err != nil {
-		return err
+	cfg.Name = strings.ToLower(strings.TrimSpace(cfg.Name))
+	if cfg.Name == "" {
+		cfg.Name, err = os.Hostname()
```

Comment:
On Linux, if you set a hostname to `MiXeD CaSe` it will remain the same, so we need to lower this one too. I think, that leading and trailing spaces are preserved too.
Hence, we should try to stick to the original `GetHostname` more closely here.

Reply:
done